pkg/server/commands/listusers/list_users_rpc.go

```diff
@@ -549,6 +573,12 @@ func (l *listUsersQuery) expandExclusion(
 		subtractFoundUsersMap[key] = struct{}{}
 	}
 
+	if baseHasCycle || subtractHasCycle {
+		return expandResponse{
+			err: nil,
+		}
+	}
```

Comment:
Can you explain in code comments why this is not

```go
if baseHasCycle || subtractHasCycle {
		return expandResponse{
		        hasCycle: true,
			err: nil,
		}
	}
```

?

Reply:
Because exclusion contains the only bespoke treatment of cycle, everywhere else we consider it a falsey outcome. Once we make a determination within the exclusion handler, we're able to properly handle the case and do not need to propagate the existence of a cycle to an upstream handler.

Reply:
> in code comments

😄 so we don't forget